public/app/core/hooks/useMediaQueryChange.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+export function useMediaQueryChange({
+  breakpoint,
+  onChange,
+  value,
+}: {
+  breakpoint: number;
+  onChange: (e: MediaQueryListEvent) => void;
+  value: boolean;
+}) {
+  useEffect(() => {
+    const mediaQuery = window.matchMedia(`(max-width: ${breakpoint}px)`);
+    const onMediaQueryChange = (e: MediaQueryListEvent) => onChange(e);
+    mediaQuery.addEventListener('change', onMediaQueryChange);
+
+    return () => mediaQuery.removeEventListener('change', onMediaQueryChange);
+  }, [breakpoint, onChange, value]);
```

Comment:
why do we need a dependency on `value`? 🤔

Reply:
Thanks @ashharrison90 , the dependency is not needed. I removed it